controllers/subscription.ts

```diff
@@ -0,0 +1,72 @@
+import { CustomRequest, CustomResponse } from "../types/global";
+const { addOrUpdate } = require("../models/users");
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const nodemailer = require("nodemailer");
+const config = require("config");
+const emailSubscriptionCredentials = config.get("emailSubscriptionCredentials");
+
+export const subscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const { email, phoneNumber } = req.body;
+    const userId = req.userData.id;
+    await addOrUpdate(
+      {
+        phoneNumber,
+        email,
+        isSubscribed: true,
+      },
+      userId
+    );
+    return res.status(201).json({
+      message: "user subscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const unsubscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const userId = req.userData.id;
+    await addOrUpdate(
+      {
+        isSubscribed: false,
+      },
+      userId
+    );
+    return res.status(200).json({
+      message: "user unsubscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const sendEmail = async (req: CustomRequest, res: CustomResponse) => {
+  try { 
+    const transporter = nodemailer.createTransport({
+      host: emailSubscriptionCredentials.host,
+      port:  emailSubscriptionCredentials.port,
+      secure: false,
+      auth: {
+        user: emailSubscriptionCredentials.user,
+        pass: emailSubscriptionCredentials.pass,
+      },
+    });
+
+    const info = await transporter.sendMail({
+      from: `"Real Dev Squad" <${emailSubscriptionCredentials.user}>`,
```

Comment:
- is this format approved by ankush?

Reply:
Not now, that is even not our priority right now, we just want to send hello on Tejas sir email as of now, our focus is to collect email and phone numbers first.

Reply:
- add a todo here to get this format approved by ankush after the email collection is working.

Reply:
https://github.com/Real-Dev-Squad/website-backend/issues/2210 i have created an issue for this sir.